soc/nordic/validate_base_addresses.c

```diff
@@ -37,6 +37,14 @@
 #define NRF_QDEC0 NRF_QDEC
 #endif
 
+#if !defined(NRF_RADIO) && defined(NRF_RADIOCORE_RADIO)
+#define NRF_RADIO NRF_RADIOCORE_RADIO
```

Comment:
The root cause, in my understanding, is that we are trying to verify peripherals from radio domain when building for other cores.
Not a blocker for me, but perhaps a cleaner solution would be to have those peripherals:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/common/nordic/nrf54h20.dtsi#L173-L296
only included in radio core builds?
Unless this is needed in some use cases that I'm not aware of.

Reply:
Pretty much. The root cause is that local peripherals are also visible to the secure domain, but it's not present upstream, so I tried to justify this patch in another way.